### ISSUE TITLE
Add content element for responsive Tabs

### ISSUE THREAD

**Issue by fkaminski** [contao/contao#3240]
A new content element for responsive tabs (to accordion) would be nice: 
For example: https://codepen.io/keenanpayne/pen/qyjrzz 

The procedure In the backend analogous to the Accordion implementation. 
Numerous solutions exist (jQuery/Vanilla JS/Pure CSS):
[Best responsive tab to accordion 2021](https://www.google.com/search?q=best+responsive+tab+to+accordion+2021) (for discussion)

**Comment by fkaminski** [comment]
I have often used this. However jQuery based:
https://github.com/samsono/Easy-Responsive-Tabs-to-Accordion

A vanilla JS sample:
https://www.gsarigiannidis.gr/tabs-on-desktop-accordion-on-mobile/
https://codepen.io/gsarig/pen/GWgNdx

**Comment by leofeyer** [comment]
I definitely want to add https://codepen.io/gsarig/pen/ExKJEoM to the core. However, we would need a lot of elements to map the structure in the back end (correct me if I‘m wrong please):

```
- tablist start
    - tabs
    - single tabpanel 1
    - tabpanel start
        - content element 1
        - content element 2
    - tabpanel end
    - single tabpanel 2
- tablist end
```

If we had #391 (nested content elements), this could be greatly simplified:

```
- tablist
    - single tabpanel 1
    - tabpanel
        - content element 1
        - content element 2
    - single tabpanel 2
```

We would not need the "end" elements and we would not need a separate "tabs" element, because the parent "tabs" element could retrieve the tab titles from its child "tabpanel" elements.

@ausi What do you think are the odds that we implement #391 anytime soon?

**Comment by m-vo** [comment]
> If we had https://github.com/contao/contao/issues/391 (nested content elements), this could be greatly simplified:

Yes, we should definitively solve nested content elements first, IMHO. This was also the reason why I left out these content elements when migrating to the fragment ones. From a data perspective there is little to no difference between tabs/accordions/sliders, sothis could simply be a rendering option or a template selection in the wrapper.

Maybe metadata for children could also be edited in the parent/wrapper. I'm thinking of a widget where there is a row/section for each child, that is added dynamically. Maybe something to discuss at the next dev meeting?

**Comment by m-vo** [comment]
By the way: I had tinkered with nested content elements some time ago. It's totally doable but IIRC there were a lot of small needed UI-wise to make it less cumbersome to work with. Plus we need to change how composition works (`PageRegular` …) and this would likely involve some well planned decision making there as well.

**Comment by ausi** [comment]
> What do you think are the odds that we implement #391 anytime soon?

I think the odds are pretty high, I’d be happy to take a shot at this in the next dev meeting.

**Comment by fkaminski** [comment]
Ping ?

**Comment by ausi** [comment]
@fkaminski #391 is implemented so there should be nothing in the way for a PR that adds tabs.
